File: samples/Mathematica/strip-comments-from-a-string.m
a = "apples, pears # and bananas
  apples, pears ; and bananas";
b = StringReplace[a, RegularExpression["[ ]+[#;].+[\n]"] -> "\n"];
StringReplace[b, RegularExpression["[ ]+[#;].+$"] -> ""] // FullForm

File: samples/Go/strip-comments-from-a-string.go
package main

import (
    "fmt"
    "strings"
)

const commentChars = "#;"

func stripComment(source string) string {
    if cut := strings.IndexAny(source, commentChars); cut >= 0 {
        return source[:cut]
    }
    return source
}

func main() {
    for _, s := range []string{
        "apples, pears # and bananas",
        "apples, pears ; and bananas",
        "no bananas",
    } {
        fmt.Printf("source:   %q\n", s)
        fmt.Printf("stripped: %q\n", stripComment(s))
    }
}
